services/web/src/App.tsx

```diff
@@ -1,16 +1,40 @@
-import { useState } from 'react';
-import reactLogo from './assets/react.svg';
-import viteLogo from '/vite.svg';
 import './App.css';
-import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useForm } from 'react-hook-form';
+import { Button } from '@/components/ui/button';
+import { Form } from '@/components/ui/form';
+import { RadioButton } from './internal_library/radioComponent';
 import { Accordion, AccordionContent, AccordionItem, AccordionTrigger } from '@radix-ui/react-accordion';
-
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
 import MockDataComponent from './website/mockComponent';
+import reactLogo from './assets/react.svg';
+import viteLogo from '/vite.svg';
+import { useState } from 'react';
+
+const FormSchema = z.object({
+    notificationPreference: z.string().min(1, { message: 'GRESHKA' }),
+});
 
-function App() {
+export function App() {
+    const form = useForm<z.infer<typeof FormSchema>>({
+        resolver: zodResolver(FormSchema),
+        defaultValues: {
+            notificationPreference: '',
+        },
+    });
+
+    const onSubmit = (data: z.infer<typeof FormSchema>) => {
+        console.log(data);
+    };
+
+    const radioOptions = [
```

Comment:
1. Move `radioOptions` constant outside of the component, because this constant is not used in any hooks. We do this because React will recreate it on each rerender. 

2. For constants that are defined outside of the component we are going to stick to using: SCREAMING_SNAKE_CASE

Reply:
It appears this is mostly for testing so that's a nit

Reply:
But it's good if you adhere to what Aleks is talking about here. That's industry standard.

Reply:
> We do this because React will recreate it on each rerender.

Essentially, React components are simple functions. If React rerenders the component, it means that the code inside the component is run once again. 

That's why if you have a state (useState) and a console.log inside a component, every time the state changes -> a rerender would be triggered and the log will be displayed in the console.

You can prevent the log from getting printed by putting inside a useEffect with an empty dependency array. But if the code is not wrapped within anything React related, it will always be rerun.

This means that any definition will be essentially reinitialized.

const something = {}; // something will be created every time the component rerenders

Why is this bad? JavaScript is a garbage collected language. Garbage collection is a really expensive process. Every time you create a new object and then stop using it, the garbage collector should mark the object as unused and collect it. Why keep creating new objects when you can just bring the object outside of the component if you know it's value is going to be constant anyways?

I hope this made sense. Feel free to ask any questions.

Reply:
Thank you for the clarifications